devices/GnssDevice/GprmcData.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using UnitsNet;
+
+namespace Iot.Device.Common.GnssDevice
+{
+    /// <summary>
+    /// Represents the GPRMC NMEA0183 data from a Gnss device. 
+    /// </summary>
+    public class GprmcData : INmeaData
+    {
+        /// <inheritdoc/>
+        public INmeaData Parse(string inputData)
+        {
+            var subfields = inputData.Split(',');
```

Comment:
Looks like this one is the only one that throws exception. I think every data should behave the same, so either we add exceptions to all or we delete it from there.
I would rather go with exceptions. Let me know what you think and we can adjust code in next PR (I can do it)

Reply:
yea, I started to add this as it can be used outside of the parser.
So, I can add everywhere.
And I kind of like the idea of adding like initially in the interface the message name. As again, this can be used directly and you can know what it is parsing. What do you think?

Reply:
Anything that can improve current implementation will be great. 
For example, check this one:
https://github.com/nanoframework/nanoFramework.IoT.Device/pull/1088/files#diff-32766b81852d3246aa0b12375d048c0c331cdcbc237bf29b9a1a781fefe8044eR104

But maybe better idea is to add static field in each class - just to reference "ID" of the data?

Reply:
I've internalised in the GnssDevice and in a GenericSerialGnssDevice all the logic.